pkg/util/limiter/memory_consumption.go

```diff
@@ -181,3 +188,22 @@ func (l *MemoryConsumptionTracker) CurrentEstimatedMemoryConsumptionBytes() uint
 
 	return l.currentEstimatedMemoryConsumptionBytes
 }
+
+// IncreaseMemoryConsumptionForLabels attempts to increase the current memory consumption based on labels.
+func (l *MemoryConsumptionTracker) IncreaseMemoryConsumptionForLabels(lbs labels.Labels) error {
+	for _, lb := range lbs {
+		// TODO: Update labels.Labels to get size of bytes directly instead of calculating it here.
```

Comment:
Are you planning to add this to `prometheus/prometheus` before merging this PR? 

If not, we'll need to modify this method to work with `stringlabels`.

Reply:
I will try to make the change on Prometheus first before merging the PR. I will try to draft the Prometheus PR today.